db/migrate/20200912190210_update_weight_calculators.rb

```diff
@@ -0,0 +1,10 @@
+class UpdateWeightCalculators < ActiveRecord::Migration
+  def change
+    Calculator::Weight.each { |calculator|
+      calculator.preferred_unit_from_list = 'kg'
+      calculator.preferred_per_unit = calculator.preferred_per_kg
+      calculator.preferences.delete(:preferred_per_kg)
+      calculator.save
+    }
+  end
```

Comment:
we also need to flush the cache keys as their values will be stale at this point. We should be able to do a `Rails.cache.delete()` passing in Spree's cache keys from https://github.com/openfoodfoundation/spree/blob/cbb24a6ed701166ffaf2ab60a402154100d74766/core/app/models/spree/preferences/configuration.rb#L31-L33 or somewhere similar. We don't want to flush the entire cache.

Reply:
So something like this? 

```
key = Calculator::Weight.new.preference_cache_key("preferred_per_kg")
Rails.cache.delete(key)
```

I did this in development, however, and it's returning nil; but maybe that's expected in dev. 
``` 
> Calculator::Weight.new.preference_cache_key("preferred_per_kg")
=> nil
```

Reply:
you can get the value of the key before your test and after to see if you are using the correct code to clear cache.

Reply:
Okay I think I understand this now. The cache key is unique to each instance of the calculator, so we'll have to do the `cache.delete` line inside the `each` block. The preference only gets written to the cache when the object gets used.

Reply:
I'd go with SQL here Andy!
This is tricky, something like this I think:
 
```ruby
def change
    Spree::Calculator.connection.execute(
      "UPDATE spree_preferences SET type = 'replace( pref.key, 'per_kg', 'per_unit')' WHERE type like '/calculator/weight/per_kg/%"
    )

    Calculator::Weight.all.each { |calculator|
      calculator.preferred_unit_from_list = 'kg'
      # preferred_per_unit vs preferred_per_kg is already taken care by the SQL above
      Rails.cache.delete("/calculator/weight/preferred_per_kg/" + calculator.id)
      calculator.save
    }
```

Reply:
clearing the cache is actually not so important as the key will be different from now on :+1: 
the new keys will be /calculator/weight/preferred_per_unit/" + calculator.id

Reply:
this will require data migration testing, can you please add to test notes above. the tester will need to go to staging before staging this PR and check some weight based payment methods and ship methods and enterprise fees. After staging this PR, these fees should all be correctly configured as before :+1:

Reply:
Nice, thanks Luis - I'll try that query out soon, and add that to the test notes.

Reply:
I updated the query a bit and pushed the changes up. That makes sense that we wouldn't need to be too concerned with clearing the cache; I left it in there since it shouldn't hurt to clean things up and I guess it'll make the cache a little smaller?